model/metadata/metadata.go

```diff
@@ -80,7 +80,10 @@ func (m *Metadata) Merge(fields common.MapStr) common.MapStr {
 	utility.Add(fields, "host", m.System.fields())
 	utility.Add(fields, "process", m.Process.fields())
 	utility.MergeAdd(fields, "service", m.Service.fields())
-	utility.MergeAdd(fields, "user", m.User.fields())
+	utility.AddIfNil(fields, "user", m.User.Fields())
```

Comment:
I don't think we should merge `user` information from `metadata` and `context.user` but overwrite it.

Reply:
@graphaelli the `user_agent` and the `client.ip` are set, even if the `context.user` information is set directly in an event, as they are nested under different root keys and not under `user`. So using `AddIfNil` seems proper here.

Reply:
Agreed on both counts.  Think we need an integration test for this too?

Reply:
The `MergeAdd` was introduced long time ago, https://github.com/elastic/apm-server/pull/687/files. 

I think it is a bug to mix two potential different `user` informations to one, that is why I changed it to `AddIfNil`, which basically first checks if there was a user sent with the specific event, and if not sets the one from the `metadata`. 

The possibility to send a `user` in `metadata` and the `event` and if this makes sense or not is a completely different topic to discuss. Please raise any concerns about this in this issue https://github.com/elastic/apm-server/issues/1255.